### PR TITLE
Case 21647: wait for snapshot signal

### PR DESCRIPTION
**Pull request by NissimHadar**
https://highfidelity.fogbugz.com/f/cases/21647/Wait-for-snapshot-signal

### CODE REVIEW

File: tests/utils/nitpick.js [hunk 1]
```diff
@@ -521,14 +542,26 @@ module.exports.runRecursive = function () {
     console.warn("Starting recursive tests");
     runningRecursive = true;
 
+    Window.stillSnapshotTaken.connect(onStillSnapshotTaken);
```
Comment (SamGondelman) [inline tests/utils/nitpick.js]: Should this be connected in onRunAuto?

File: tests/utils/nitpick.js [hunk 2]
```diff
@@ -521,14 +542,26 @@ module.exports.runRecursive = function () {
...
+                if (quitRequested) {
+                    console.warn("Quitting due to request");
+                    Script.stop();
+                }
+
                 if (testCases.length > 0) {
                     currentTestCase = testCases.pop();
-                    runOneTestCase(currentTestCase, testMode);
+
+                    if (!quitRequested) {
```
Comment (SamGondelman) [inline tests/utils/nitpick.js]: I don't think `quitRequested` could be set in between line 554 and here, so this might not be necessary.

Comment (NissimHadar) [reply tests/utils/nitpick.js]: That is what I thought.  But - I think that the script does not stop immediately, and the effect is that when the script does stop - some entities have been created.

File: tests/utils/nitpick.js (hunk 1 above)
Comment (NissimHadar) [reply tests/utils/nitpick.js]: Yes - that is a better idea.

File: tests/utils/nitpick.js (hunk 2 above)
Comment (SamGondelman) [reply tests/utils/nitpick.js]: Perhaps it should `return` after `Script.stop()`?

File: tests/utils/nitpick.js [hunk 3]
```diff
@@ -490,6 +511,7 @@ module.exports.addStepSnapshot = function (name, stepFunction) {
 // The default time between test steps may be modified through these methods
 module.exports.enableAuto = function () {
     testMode = "auto";
+    Window.stillSnapshotTaken.connect(onStillSnapshotTaken);
```
Comment (SamGondelman) [inline tests/utils/nitpick.js]: This still seems wrong...why isn't it next to where you connect the key press event in `onRunAuto`?

Comment (NissimHadar) [reply tests/utils/nitpick.js]: onRunAuto isn't called when running a recursive script.

Comment (SamGondelman) [reply tests/utils/nitpick.js]: ah ok

File: tests/utils/nitpick.js (hunk 2 above)
Comment (NissimHadar) [reply tests/utils/nitpick.js]: Yep - that works

**Pull request merged**